packages/-ember-data/tests/integration/relationships/belongs-to-test.js

```diff
@@ -119,6 +119,99 @@ module('integration/relationship/belongs-to BelongsTo Relationships (new-style)'
     assert.ok(personPet === pet, 'We ended up in the same state');
   });
 
+  test('async belongsTo resolves to the record when sideloaded', async function(assert) {
+    let personFindRecordCalls = 0;
+    this.owner.register(
+      'adapter:pet',
+      JSONAPIAdapter.extend({
+        shouldBackgroundReloadRecord() {
+          return false;
+        },
+        findRecord() {
+          assert.ok(false, 'We should not attempt to fetch the pet');
+          // still resolve so that test can continue nicely
+          return resolve({
+            data: {
+              type: 'pet',
+              id: '1',
+              attributes: { name: 'Shen' },
+              relationships: {
+                bestHuman: {
+                  data: { type: 'person', id: '1' },
+                },
+              },
+            },
+          });
+        },
+        findBelongsTo() {
+          assert.ok(false, 'We should not attempt to fetch the person for out pet');
+          return this.store.adapterFor('person').findRecord();
+        },
+      })
+    );
+
+    this.owner.register(
+      'adapter:person',
+      JSONAPIAdapter.extend({
+        shouldBackgroundReloadRecord() {
+          return false;
+        },
+        findRecord() {
+          assert.equal(++personFindRecordCalls, 1, 'We call findRecord only once for our person');
+          return resolve({
+            data: {
+              type: 'person',
+              id: '1',
+              attributes: { name: 'Chris' },
+              relationships: {
+                bestDog: {
+                  data: { type: 'pet', id: '1' },
+                  links: {
+                    related: './pet/1',
```

Comment:
Didn't know we supported this format

Reply:
Yes... We support jsonapi links for relationships and this is the more common format: https://jsonapi.org/format/#document-links